app/javascript/components/map-menu/components/sections/my-gfw/component.jsx

```diff
@@ -0,0 +1,256 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import cx from 'classnames';
+
+import MyGFWLogin from 'components/mygfw-login';
+import Button from 'components/ui/button/button-component';
+import Dropdown from 'components/ui/dropdown';
+import Icon from 'components/ui/icon/icon-component';
+import Pill from 'components/ui/pill';
+import editIcon from 'assets/icons/edit.svg';
+import tagIcon from 'assets/icons/tag.svg';
+import subscribedIcon from 'assets/icons/subscribed.svg';
+import logoutIcon from 'assets/icons/logout.svg';
+import screenImg1x from 'assets/images/aois/single A.png';
+import screenImg2x from 'assets/images/aois/single A @2x.png';
+
+import './styles.scss';
+
+class MapMenuMyGFW extends PureComponent {
+  constructor() {
+    super();
+    this.state = {
+      activeTags: {}
+    };
+  }
+
+  setTags() {
+    const { areas } = this.props;
+    const tags = {};
+    if (areas) {
+      areas.forEach(area =>
+        area.tags.forEach(tag => {
+          if (tags[tag] === undefined) {
+            tags[tag] = false;
+          }
+        })
+      );
+    }
+    this.setState({ activeTags: tags });
+  }
+
+  componentDidMount() {
+    this.setTags();
+  }
+
+  renderLoginWindow() {
```

Comment:
You could make each of these section components to make this file simpler. Up to you, but just a thought as there are more than two renderAlgo() funcs.